src/libraries/System.Runtime.Loader/tests/System.Runtime.Loader.Tests.csproj

```diff
@@ -43,4 +45,19 @@
 
     <TrimmerRootDescriptor Include="$(MSBuildThisFileDirectory)ILLink.Descriptors.xml" />
   </ItemGroup>
+
+  <Target Name="IncludeDeltasInWasmBundle"
+	  BeforeTargets="PrepareForWasmBuildApp"
+	  Condition="'$(TargetOS)' == 'Browser'">
+    <ItemGroup>
+      <!-- FIXME: this belongs in eng/testing/tests.wasm.targets -->
+      <!-- FIXME: Can we do something on the Content items in the referenced projects themselves to get this for free? -->
+      <WasmFilesToIncludeInFileSystem Include="@(PublishItemsOutputGroupOutputs)"
+				      Condition="$([System.String]::new('%(PublishItemsOutputGroupOutputs.Identity)').EndsWith('.dmeta'))" />
+      <WasmFilesToIncludeInFileSystem Include="@(PublishItemsOutputGroupOutputs)"
+				      Condition="$([System.String]::new('%(PublishItemsOutputGroupOutputs.Identity)').EndsWith('.dil'))" />
+      <WasmFilesToIncludeInFileSystem Include="@(PublishItemsOutputGroupOutputs)"
+				      Condition="$([System.String]::new('%(PublishItemsOutputGroupOutputs.Identity)').EndsWith('.dpdb'))" />
+    </ItemGroup>
```

Comment:
IIUC:

- `@(Content)` is added in project (`*ApplyUpdate*`), and the main test project references that
- `tests.wasm.targets` adds	`@(ContentWithTargetPath)` to the wasm vfs
  - but that gets populated only by the `@(Content)` items from the test project itself
  - Out of all full set of copy-local items from referenced projects, I couldn't figure out a way to differentiate the ones that came from `@(Content)`[1]
  - it wouldn't make sense to add *all* the copy-local files to the vfs though!

- It would be nice to have some idiom for what gets automatically added to the vfs, and maybe `WasmApp.targets` can use that
- Maybe, XI/XA have a solution for this? @rolfbjarne @jonathanpeppers 

- Otherwise, I think, at least for now, your approach is fine 😬 

--
1. um there is a property that would allow retaining the original metadata from the `Content`, so that could be helpful here, but it sounds hacky!

Reply:
> um there is a property that would allow retaining the original metadata from the `Content`, so that could be helpful here, but it sounds hacky!

To be honest filtering by metadata sounds better to me than going by file extensions.

If I put `DeltaOutputType` metadata on the `Content` in the references (it's already there on the `_HotReloadDeltaGeneratorOutputs`, but I forgot to tack on the `KeepMetadata` thing), then, hopefully, that will propagate to the parent project's `PublishItemsOutputGroupOutputs`?
And then I would feel less bad about adding them to `WasmFilesToIncludeInFileSystem` in `eng/testing/tests.wasm.targets`.

Reply:
On Android we don't actually use `@(Content)` at all and emit warnings. This is just because of the way `.apk` files are compressed on disk, we can't put a file somewhere where `File.Open()` would even work.

You might be able to do something like this to grab files from all `@(ProjectReference)`, though:

```xml
<MSBuild Projects="@(ProjectReference)" Targets="HotReloadDeltaGeneratorContentForOutputs">
  <Output TaskParameter="TargetOutputs" ItemName="_ContentFromProjectReference" />
</MSBuild>
```

Not sure if that helps or not.

Reply:
I'm not quite sure I understand what you need to do. It seems that you're trying to build an app that contains both the original assemblies + the modified assemblies that should be loaded through hot reload, and then those modified assemblies should be in a specific location within the app bundle, is that right?